controllers/certificaterequest_controller.go

```diff
@@ -161,6 +174,64 @@ func (r *CertificateRequestReconciler) SetupWithManager(mgr ctrl.Manager) error
 // 	return false
 // }
 
+// requestShouldBeProcessed will return false if the conditions on the request
+// mean that it should not be processed. If the request has been denied, it
+// will set the request failure time and add a Ready=False condition.
+func (r *CertificateRequestReconciler) requestShouldBeProcessed(ctx context.Context, log logr.Logger, cr *cmapi.CertificateRequest) (bool, error) {
```

Comment:
Are you planning to add a test for this function or do you think it is not necessary?

Reply:
Wasn't planing to but happy to add some tests in.

Reply:
It would be great, thanks.